pkg/humio/client.go

```diff
@@ -196,3 +214,74 @@ func (c *Client) Fetch(method string, path string, body *bytes.Buffer, out inter
 	}
 	return fmt.Errorf("%s %s", res.Status, strings.TrimSpace(errResponse.Detail))
 }
+
+func (c *Client) Stream(method string, path string, query Query, ch chan StreamingResults, done chan any) error {
+	var humioQuery struct {
+		QueryString string `json:"queryString"`
+		Start       string `json:"start,omitempty"`
+		Live        bool   `json:"isLive"`
+	}
+	humioQuery.QueryString = query.LSQL
+	humioQuery.Start = query.Start
+	humioQuery.Live = true
+
+	defer close(done)
+
+	var buf bytes.Buffer
+	err := json.NewEncoder(&buf).Encode(humioQuery)
+	if err != nil {
+		return err
+	}
+	url, err := url.JoinPath(c.URL.String(), path)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest(method, url, &buf)
+	if err != nil {
+		return err
+	}
+	req = c.addAuthHeaders(req)
+
+	// Create a context with a timeout to avoid endless streaming
+	ctx, cancel := context.WithTimeout(context.Background(), 200*time.Second)
+	defer cancel()
+
+	req = req.WithContext(ctx)
+
+	res, err := c.StreamingClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		res.Body.Close()
+	}()
+
+	d := json.NewDecoder(res.Body)
+
+	ticker := time.NewTicker(1 * time.Second)
```

Comment:
More of a curiosity question - why have we chosen 1 second? Just an arbitrary choice? Again, potentially something that could be configurable?

Reply:
Yes - arbitrary choice and something we could make configurable.